### PR TITLE
incompatibility-revisions

### PR DESCRIPTION
**Pull request by Raulp8**
- adds revisions to incompatibility package including updating classifications and fixing file structure

### CODE REVIEW

File: incompatibility/incomp-scanning_test.go [hunk 1]
```diff
@@ -85,7 +87,7 @@ func TestIncompatibilityExistence(t *testing.T) {
 				_, searchErr :=
 					findNode(node.Content[0], incomp.GetTokenPath()...)
 				if searchErr != nil {
-					tt.Errorf(searchErr.Error())
+					tt.Errorf(searchErr.Error(), fmt.Sprintf("%+v\n", incomp))
```
Comment (skeid21) [inline incompatibility/incomp-scanning_test.go:90]: Do we want this in the form of want, got?

File: incompatibility/incomp-scanning_test.go [hunk 2]
```diff
@@ -67,9 +68,10 @@ func TestIncompatibilityExistence(t *testing.T) {
 	var existenceTest = []struct {
 		path string
 	}{
-		{"../../examples/petstore/petstore.yaml"},
-		{"../oas-examples/petstore.json"},
-		{"../../examples/bookstore/bookstore.yaml"},
+		{"../examples/petstore/petstore.yaml"},
```
Comment (skeid21) [inline incompatibility/incomp-scanning_test.go:71]: I don't know that I fully follow what you are testing with this test?  It seems that you intend to test at least one incompatibility is found and that you can map that to a node in the yaml.  

But you aren't verifying any specific incompatibilities are being found which is what the method names seems to imply to me.

File: incompatibility/reporter.go [hunk 3]
```diff
@@ -143,6 +143,9 @@ func ComponentsSearch(doc *openapiv3.Document) []*Incompatibility {
 			)
 		}
 	}
+	if doc.Components.SecuritySchemes != nil {
```
Comment (skeid21) [inline incompatibility/reporter.go:146]: I don't see a test for this new case.

**Review by skeid21 — CHANGES_REQUESTED** [review]

File: incompatibility/reporter.go (hunk 3 above)
Comment (skeid21) [reply incompatibility/reporter.go:146]: I saw your message from the other PR that you now have a spec file which has this.  But we aren't actually testing that this case is found.    See the comment I left in incomp-scanning_tests.go.  Let me know if you have questions or want to chat.

File: incompatibility/incomp-scanning_test.go (hunk 1 above)
Comment (Raulp8) [reply incompatibility/incomp-scanning_test.go:90]: think reporting the missing incompatibility is enough, this test doesn't really include a want, what we are looking for here is that there exists such a node associated with the reported path in the incompatibility

File: incompatibility/reporter.go (hunk 3 above)
Comment (Raulp8) [reply incompatibility/reporter.go:146]: leaving ticket to indicate future testing is need see [here](https://b.corp.google.com/issues/195057972)

Comment (skeid21) [reply incompatibility/reporter.go:146]: ack

**Review by skeid21 — APPROVED** [review]
Per discussion in person.  More test coverage to come in a future PR

**Pull request merged**